lib/src/kbasesearchengine/search/ElasticIndexingStorage.java

```diff
@@ -637,6 +637,11 @@ private int updateLastVersionsInData(String indexName, GUID parentGUID,
         if (indexName == null) {
```

Comment:
Another question - indexObjects uses checkIndex() to get the index name, which returns an index name with a version. Then it calls loadLastVersion() with that index name. That seems to imply that the last version loaded could be incorrect if the higher version number was in an index different from that to which the data was eventually indexed.

Reply:
The next few lines after the referenced line above replace the version with a wild card in order to update all indexes with the specified index name.

Reply:
That's not what I mean. The steps are:

1) indexObjects calls checkIndex
2) checkIndex returns an index name like MyType_2, where 2 is the version of the search type MyType
3) indexObjects calls loadLastVersion()
4) loadLastVersion() looks up the maximum version of the object in index MyType_2 and returns it

that's where we run into trouble, because there's no guarantee that the last version of the object is in the MyType_2 index. It might be in MyType_1, MyType_3, or whatever. So loadLastVersion() might return the wrong maximum version, which would cause islast to be set incorrectly when the versions are updated after the objects are saved.

Reply:
I think you are right. I have changed loadLastVersion() to return the global last version and not the local last version.

Reply:
Should add a test that exercises this scenario - I think just indexing a high object version in with a search spec with version X and then a low object version with a search spec with version Y will do it. Without the fix you'd expect the low version to be set with islast=true.

Reply:
You might also want to make the index name munging code a function, as it looks like it's duplicated in two places.

Reply:
Done.